errors.go

```diff
@@ -18,3 +18,38 @@ func (e errorList) Error() string {
 	}
 	return "multiple errors:\n" + strings.Join(msgs, "\n")
 }
+
+// IgnorableError is an error that has an extra method to tell whether it should
+// be ignored. Ignored errors may be omitted from standard program output.
+type IgnorableError interface {
+	error
+	Ignore() bool
+}
+
+// ignoredError implements IgnorableError, always returning true.
+type ignoredError struct {
+	err error
+}
+
+func (e *ignoredError) Error() string {
+	if e.err == nil {
+		return ""
+	}
+	return e.err.Error()
+}
+
+func (e *ignoredError) Ignore() bool {
```

Comment:
I guess both words are a bit confusing, I feel like `error.Ignore()` is instructing the error to be ignore-able (like I'd imagine `error.Ignore(true)`). How about `CanIgnore`?